### PR TITLE
Add Tithe Farm plugin

### PR DESCRIPTION
**Pull request by Unmoon**
Features:
- Timers on plants
- Inventory overlay for watering can doses
- Replaces the official sack overlay with a Runelite standard overlay

https://www.youtube.com/watch?v=Q9Au6b8-Vao
image

Known issue: Ghost timers appear if you get a loading screen while walking in the Tithe Farm. It's an issue with onGameObjectsChanged I think, ~~but can make a workaround if needed.~~ I tried some workarounds, but none worked.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+import net.runelite.api.Client;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.awt.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java]: Don't use star imports

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmInventoryOverlay.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import net.runelite.api.queries.InventoryWidgetItemQuery;
+import net.runelite.api.widgets.WidgetItem;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.TextComponent;
+import net.runelite.client.util.QueryRunner;
+
+import javax.inject.Inject;
+import java.awt.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmInventoryOverlay.java]: Don't use star imports

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	private void checkPlants()
+	{
+		Set<TitheFarmPlant> plantsToRemove = new HashSet<>();
+		for (TitheFarmPlant plant : plugin.getPlants())
+		{
+			if (plant.getPlantTimeRelative() == 1)
+			{
+				plantsToRemove.add(plant);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java]: Why are you not calling plugin.removePlant(plant) directly here?

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPluginConfig.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+			name = "Show watering can doses",
+			description = "Configures whether or not the watering can doses are displayed"
+	)
+	default boolean showWateringCanOverlay()
+	{
+		return true;
+	}
+
+	@ConfigItem(
+			position = 2,
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPluginConfig.java]: Make the indentation for the ConfigItem fields consistent withing the class

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java (hunk 3 above)
Comment (Unmoon) [reply runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java]: It creates warnings about changing the set while iterating through it. If it's okay to get those warnings I can change it back.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java]: plugin.getPlants().removeIf(plant -> plant.getPlantTimeRelative() == 1)

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmInventoryOverlay.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+class TitheFarmInventoryOverlay extends Overlay
+{
+	private final QueryRunner queryRunner;
+	private final TitheFarmPluginConfig config;
+
+	@Inject
+	TitheFarmInventoryOverlay(QueryRunner queryRunner, TitheFarmPluginConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmInventoryOverlay.java]: After the update to change default overlay layer, this overlay layer should be set to ABOVE_WIDGETS

File: runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+	private Color colorUnwatered;
+	private Color colorWatered;
+	private Color colorGrown;
+
+	@Inject
+	TitheFarmPlantOverlay(@Nullable Client client, TitheFarmPlugin plugin, TitheFarmPluginConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java]: After the change to the default overlay layer and addition of new layer, this layer should be set to ABOVE_SCENE

**Pull request merged**